chia/data_layer/data_store.py

```diff
@@ -720,6 +722,7 @@ async def insert(
                     left = reference_node_hash
                     right = new_terminal_node_hash
 
+                assert len(ancestors) <= 65
```

Comment:
looks like leftover code I used to test autoinsert balances the tree - I'm happy to remove it, but this was the sanity check needed to see we balance properly. the height will turn to `len(ancestors) + 1`, so if this holds for every insert we are guaranteed the whole tree will be balanced. probably it'll be good to remove if we don't want to enforce the 62 max-height condition in the future.

Reply:
My understanding is that in so much as asserts belong in runtime code they should indicate fairly localized programming errors.  A large depth could be created by adding data which is not a programming error.  It would just be exceeding a condition of what we support.  Raising an exception clarifying what is going on seems better to me.  I also commented in keybase for a more general discussion of assertions.